Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -631,7 +631,9 @@ void FieldProbe::ComputeDiags (int step)
             amrex::Vector<int> displs_vector;
             if (amrex::ParallelDescriptor::IOProcessor()) {
                 displs_vector.resize(mpisize, 0);
-                displs_vector[0] = 0;
+                if (mpisize > 0) {
```

Comment:
Shouldn't `mpisize` be at least `1` even in serial runs?

Reply:
Yes, we know that, but the compiler doesn't. It was giving a warning that `dispels_vector` is uninitialized. That is one way to remove the warning. Is there a different way?

Reply:
Got it. We could call https://en.cppreference.com/w/cpp/container/vector/assign after resize, even if we technically only need to ensure the value for the first element is zero.

Reply:
Looking at it again, that line `displs_vector[0] = 0;` shouldn't be needed at all since the vector is already initialized to `0` in the `resize`.

Reply:
No, resize only sets new elements to the value provided.

Reply:
Yes, but here `displs_vector` is declared as an empty vector at line 631 so all elements are new.